enterprise/coderd/coderd_test.go

```diff
@@ -0,0 +1,193 @@
+		t.Parallel()
+		client := coderdenttest.New(t, nil)
+		_ = coderdtest.CreateFirstUser(t, client)
+		license := coderdenttest.AddLicense(t, client, coderdenttest.LicenseOptions{
+			UserLimit: 100,
+			AuditLog:  true,
+		})
+		res, err := client.Entitlements(context.Background())
+		require.NoError(t, err)
+		assert.True(t, res.HasLicense)
+		al := res.Features[codersdk.FeatureAuditLog]
+		assert.Equal(t, codersdk.EntitlementEntitled, al.Entitlement)
+		assert.True(t, al.Enabled)
+
+		err = client.DeleteLicense(context.Background(), license.ID)
+		require.NoError(t, err)
+
+		res, err = client.Entitlements(context.Background())
+		require.NoError(t, err)
+		assert.False(t, res.HasLicense)
+		al = res.Features[codersdk.FeatureAuditLog]
+		assert.Equal(t, codersdk.EntitlementNotEntitled, al.Entitlement)
+		assert.True(t, al.Enabled)
+	})
+	t.Run("Warnings", func(t *testing.T) {
+		t.Parallel()
+		client := coderdenttest.New(t, nil)
+		first := coderdtest.CreateFirstUser(t, client)
+		for i := 0; i < 4; i++ {
+			coderdtest.CreateAnotherUser(t, client, first.OrganizationID)
+		}
+		coderdenttest.AddLicense(t, client, coderdenttest.LicenseOptions{
+			UserLimit: 4,
+			AuditLog:  true,
+			GraceAt:   time.Now().Add(-time.Second),
+		})
+		res, err := client.Entitlements(context.Background())
+		require.NoError(t, err)
+		assert.True(t, res.HasLicense)
+		ul := res.Features[codersdk.FeatureUserLimit]
+		assert.Equal(t, codersdk.EntitlementGracePeriod, ul.Entitlement)
+		assert.Equal(t, int64(4), *ul.Limit)
+		assert.Equal(t, int64(5), *ul.Actual)
+		assert.True(t, ul.Enabled)
+		al := res.Features[codersdk.FeatureAuditLog]
+		assert.Equal(t, codersdk.EntitlementGracePeriod, al.Entitlement)
+		assert.True(t, al.Enabled)
+		assert.Nil(t, al.Limit)
+		assert.Nil(t, al.Actual)
+		assert.Len(t, res.Warnings, 2)
+		assert.Contains(t, res.Warnings,
+			"Your deployment has 5 active users but is only licensed for 4.")
+		assert.Contains(t, res.Warnings,
+			"Audit logging is enabled but your license for this feature is expired.")
+	})
+	t.Run("Pubsub", func(t *testing.T) {
+		t.Parallel()
+		client, _, api := coderdenttest.NewWithAPI(t, nil)
+		entitlements, err := client.Entitlements(context.Background())
+		require.NoError(t, err)
+		require.False(t, entitlements.HasLicense)
+		coderdtest.CreateFirstUser(t, client)
+		_, err = api.Database.InsertLicense(context.Background(), database.InsertLicenseParams{
+			UploadedAt: database.Now(),
+			Exp:        database.Now().AddDate(1, 0, 0),
+			JWT: coderdenttest.GenerateLicense(t, coderdenttest.LicenseOptions{
+				AuditLog: true,
+			}),
+		})
+		require.NoError(t, err)
+		err = api.Pubsub.Publish(coderd.PubsubEventLicenses, []byte{})
+		require.NoError(t, err)
+		require.Eventually(t, func() bool {
+			entitlements, err := client.Entitlements(context.Background())
+			assert.NoError(t, err)
+			return entitlements.HasLicense
+		}, testutil.WaitShort, testutil.IntervalFast)
```

Comment:
the previous tests included what happens when multiple licenses are added, but this one includes only one.